### PR TITLE
Propagate a lazy proxy of the storage model

### PR DESCRIPTION
**Pull request by M4rtinK**
The storage model for the partitioning modules should be always created lazily. When we reset the partitioning, the new model shouldn't be created until we try to work with it. Then we create a new copy of the storage model, hide disks that are not selected and possibly initialize empty disks.

There is a problem with modules that need to be able to work with the same copy of the storage model as the partitioning modules. At this moment, it is only the device tree module. The suggested solution is to propagate a lazy proxy of the storage model. It will not trigger the creation of the copy until we try to access the attributes of the storage model.

Basically, the device tree module always gets the storage model on demand from the storage property of the partitioning module.

(cherry picked from commit 39d3a894411e3069cdb14354509153028a48e6c5)

Related: rhbz#1868577
Resolves: RHEL-16276

### CODE REVIEW

**Comment by M4rtinK** [comment]
Cherry pick from upstream with a possible fix for RHEL-16276. Not yet tested, but opening a draft PR for general spoke testing via our CI first.

**Review by VladimirSlavik — COMMENTED** [review]
Cherry-picked code lgtm.

**Comment by M4rtinK** [comment]
I've both reproduced the issue without this fix on RHEL 8 and also checked that if the fix is added, the issue no longer manifests. So marking this as ready for review. :)
